### DISCUSSION TITLE
is there scenario-level variable

### DISCUSSION THREAD

**Discussion by bodycombathuang** [artilleryio/artillery#2335] [Artillery]
I have more questions
1. I know we can specify variables under "config",  is there scenario-level variable?
2. can multiple functions be called in one scenario?
scenarios:
  - name: "scenario1"
    flow:
      - function "function1"
      - function "function2"
  3. can I specify different variable for different function under 1 scenario?

**Comment by bodycombathuang** [comment]
hi @bernardobridge , could you help me, thank you very much

**Comment by bernardobridge** [comment]
👋 

>1. I know we can specify variables under "config", is there scenario-level variable?

There are no scenario-level variables. They are set within the `config`. You can also use the `--variable` flag to [set them on the fly](https://www.artillery.io/docs/reference/cli/run#set-scenario-variables-on-the-fly), or use a `beforeScenario` [hook](https://www.artillery.io/docs/reference/engines/http#setting-scenario-level-hooks) to set them via Javascript for the scenario (`context.vars.varName = varValue`)

---

> 2. can multiple functions be called in one scenario?

Yes. Documentation: https://www.artillery.io/docs/reference/engines/http#function-steps-in-a-flow

---

> 3. can I specify different variable for different function under 1 scenario?

I don't understand the question here. As I mentioned to you in another thread, functions don't take arguments in the YAML. You can access variables via `context.vars.varName` within them, so any variable you have set somewhere else (`config`, flags, or another hook) will be available then.

**Comment by bodycombathuang** [comment]
hi @bernardobridge if a scenario calls multiple functions , I want each functions to use one variable, but the value of the variable is different for all functions, but if variable defined in the config is accessed by all functions in a scenario, the value of the varaible will be same for all the functions

**Comment by bodycombathuang** [comment]
scenarios:
    - name: "scenario1"
      flow:
        - post:
            url: "/automation"
            json:
              tag: "tag1"
            name: "automation1"
        - post:
            url: "/automation"
            json:
              tag: "tag2"
            name: "automation2"  

you can see here, 2 requests use "tag" property, but values are  different , one is tag1 and another is tag2

but if I use -function to replace http call, function has no argument ,
scenarios:
    - name: "scenario1"
      flow:
        - function: "function1"
        - function : "function1"
       I cannot achieve that, in the first funciton call, "tag1" value of tag property  is passed, in the second function call, "tags" is passed  

hi @bernardobridge could you help me, thank you
